resource/modules/resource_match.cpp

```diff
@@ -614,59 +629,56 @@ static inline bool is_existent_jobid (const resource_ctx_t *ctx, uint64_t jobid)
 
 static int run_remove (resource_ctx_t *ctx, int64_t jobid)
 {
-    int rc = 0;
+    int rc = -1;
     dfu_traverser_t &tr = *(ctx->traverser);
-    if ((rc = tr.remove (jobid)) == 0) {
-        if (is_existent_jobid (ctx, jobid)) {
-           job_info_t *info = ctx->jobs[jobid];
-           info->state = job_lifecycle_t::CANCELLED;
-        }
-    } else {
+
+    if ((rc = tr.remove (jobid)) < 0) {
         if (is_existent_jobid (ctx, jobid)) {
+           // When this condition arises, we will be less likley
```

Comment:
```suggestion
           // When this condition arises, we will be less likely
```

I'm not sure I'm parsing the first sentence of this comment correctly.  Where does the probability/likelihood in jobid reuse come in?  I thought that was a deterministic process.

Reply:
At the moment, I can't find the commit message. But a same jobid will have to be used as it often has to be reserved multiple times before it will finally be allocated.